Assets/Scripts/BossController.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class BossController : MonoBehaviour {
+    // Pubic Members
+    public float health = 500f; // Starting health value
+    public float movementSpeed = 8f; // Speed enemy will move through the world
+    public float attackRange = 20f; // Max distance you can be from player to attack
+    public float attackStrength = 15f; // Damage that will be done to player upon successful attack
+    public Transform playerObject; // Player transform object
+    
+    // Private Members
+    private Rigidbody2D _body; // Enemy rigidbody, will get from object
+    private Vector2 _playerDirection; // Direction player is relative to player, used for tracking and following player
+    private float _playerDistance;
+
```

Comment:
// Property for Health
    public float Health {
        get { return _health; }
        set {
            _health = Mathf.Clamp(value, 0, maxHealth);
            OnHealthChanged();
            if (_health <= 0) {
                Die();
            }
        }
    }